libs/logic-apps-shared/src/designer-client-services/lib/standard/connection.ts

```diff
@@ -321,14 +321,15 @@ export class StandardConnectionService extends BaseConnectionService implements
     const {
       apiHubServiceDetails: { tenantId },
       workflowAppDetails,
+      baseUrl
     } = this._options;
     if (!isArmResourceId(connection.id) || !workflowAppDetails) {
       return;
     }
 
     const intl = getIntl();
 
-    if (!isIdentityAssociatedWithLogicApp(workflowAppDetails.identity)) {
+    if (!isHybridLogicApp(baseUrl) && !isIdentityAssociatedWithLogicApp(workflowAppDetails.identity)) {
```

Comment:
can we create a separate helper function to validate identity validity?

something like 
```
private validateLogicAppIdentity(baseUrl: string, identity: Identity | undefined) {
  if (!isHybridLogicApp(baseUrl) && !isIdentityAssociatedWithLogicApp(identity)) {
    throw new Error(
      intl.formatMessage({
        defaultMessage: 'A managed identity is not configured on the logic app.',
        id: 'WnU9v0',
        description: 'Error message when no identity is associated',
      })
    );
  }

  if (isHybridLogicApp(baseUrl)) {
    if (!identity?.principalId || !identity?.tenantId) {
      throw new Error(
        intl.formatMessage({
          defaultMessage: 'App identity is not configured on the logic app environment variables.',
          id: 'WnU9v0',
          description: 'Error message when no app identity is added in environment variables',
        })
      );
    }
  }
}
```